base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1169,7 +1169,9 @@ fn handle_possible_reorg<T: BlockchainBackend>(
             );
             Err(e)
         })?;
-    insert_orphan(db, block.clone())?;
+    if !db.contains(&DbKey::OrphanBlock(block.hash()))? {
```

Comment:
This extra check does not need to be performed as the backend will handle the case of a duplicate orphan block.